library/Fission/Web/Domain.hs

```diff
@@ -0,0 +1,31 @@
+module Fission.Web.Domain
+  ( API
+  , server
+  ) where
+
+import RIO
+import Data.Has
+
+import           Servant
+
+import           Fission.Web.Server
+import           Fission.User        as User
+import           Fission.IPFS.CID.Types
+
+import qualified Network.AWS.Auth    as AWS
+import qualified Fission.AWS.Types   as AWS
+import qualified Fission.AWS.Route53 as Route53
+
+type API = Capture "cid" CID
+        :> Post    '[PlainText, OctetStream] AWS.Domain
```

Comment:
This should be a `PUT`, no? We have the ID, we're not creating a new one

Reply:
Well we're upserting a DNS record. So in some cases we're creating a new DNS record

Reply:
Sure, but that's a VERY common misconception about `PUT` and `POST`

> It's quite possible, valid and even preferred in some occasions, to use PUT to create resources, or use POST to update resources.

[Source](http://restcookbook.com/HTTP%20Methods/put-vs-post/)

* `POST`
  * You don't have a primary key yet
  * Calling multiple times may yield different results
* `PUT`
  * You know the key of the thing you're updating
  * Idempotent

It's not about of the record has been persisted previously; it's about how you reference the thing you're talking about. We know the key of the DNS record: it's their username! I believe that this would be idempotent, too, no?

A little reference table:

HTTP Method | Idempotent | Safe
-- | -- | --
OPTIONS | yes | yes
GET | yes | yes
HEAD | yes | yes
PUT | yes | no
POST | no | no
DELETE | yes | no
PATCH | no | no

[Source](http://restcookbook.com/HTTP%20Methods/idempotency/)

Reply:
The more you know! Yup should definitely be PUT